### ISSUE TITLE
Build challenges inside a Docker container

### ISSUE THREAD

**Issue by C0deH4cker** [C0deH4cker/PwnableHarness#3]
By performing builds inside of a Docker container, it would ensure consistent builds that don't depend on environment external to the repo, and challenges can be built on platforms other than Linux.

**Comment by C0deH4cker** [comment]
As of #6, challenges can now be built and run on macOS. This is different than the goal of this issue, however there is some overlap in the use cases.

**Comment by C0deH4cker** [comment]
This is now more important. When challenges are compiled on newer versions of Ubuntu like 22.04, then we try to run them in this Ubuntu 16.04 container, it fails with this error message:

```
/usr/local/bin/pwnableserver: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /usr/local/bin/pwnableserver)
```

This is because the glibc version outside of the container that the challenge binary was linked against is much newer than the version present in the container, resulting in a version mismatch error.

**Comment by C0deH4cker** [comment]
Implemented in #16
